### PR TITLE
Place floats in layout 2020, but don't flow text around the floats yet.

### PR DESCRIPTION
**Pull request by pcwalton**
This commit puts floats behind the `layout.floats.enabled` pref, because of the
following issues and unimplemented features:

* Inline formatting contexts don't take floats into account, so text doesn't
  flow around the floats yet.

* Non-floated block formatting contexts don't take floats into account, so BFCs
  can overlap floats.

* The `clear` property is only respected on floats, not on other elements; i.e.
  regular boxes can't clear past floats yet.

* Block formatting contexts that contain floats don't expand vertically to
  contain all the floats. That is, floats can stick out the bottom of BFCs,
  contra spec.

* Floats don't respect negative margins yet.

* Floats are painted in the same order as regular content, contra spec.

r? @Manishearth

---

- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because ___

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]
r=me

File: tests/wpt/mozilla/meta-layout-2020/mozilla/localeCompare.html.ini [hunk 1]
```diff
@@ -1,4 +1,2 @@
 [localeCompare.html]
   bug: https://github.com/servo/servo/issues/25802
-  [localeCompare should return the same as other browsers, even though it's implementation-dependent]
-    expected: FAIL
```
Comment (jdm) [inline tests/wpt/mozilla/meta-layout-2020/mozilla/localeCompare.html.ini:4]: This needs to be reverted.

File: tests/wpt/mozilla/meta-layout-2020/mozilla/navigator.html.ini [hunk 2]
```diff
@@ -1,4 +1,4 @@
 [navigator.html]
-  [navigator.platform mac]
+  [navigator.platform linux]
```
Comment (jdm) [inline tests/wpt/mozilla/meta-layout-2020/mozilla/navigator.html.ini:2]: This needs to be reverted.

File: tests/wpt/mozilla/meta-layout-2020/mozilla/promise.html.ini [hunk 3]
```diff
@@ -0,0 +1,19 @@
+[promise.html]
+  [Resolve callback gets argument]
```
Comment (jdm) [inline tests/wpt/mozilla/meta-layout-2020/mozilla/promise.html.ini:2]: These failures are very surprising.

**Pull request closed without merging**